core/src/tangram.cpp

```diff
@@ -90,16 +98,21 @@ static std::bitset<8> g_flags = 0;
 Map::Map() {
 
     impl.reset(new Impl());
+    impl->valid.store(true);
 
 }
 
 Map::~Map() {
     // The unique_ptr to Impl will be automatically destroyed when Map is destroyed.
+    impl->valid.store(false);
     TextDisplay::Instance().deinit();
     Primitives::deinit();
 }
 
 void Map::Impl::setScene(std::shared_ptr<Scene>& _scene) {
+
+    if (!valid.load()) { return; }
+
     {
```

Comment:
I think the `valid` check is prone to races - why calling `impl->setScene` when `impl` may be invalid here?